pkg/operator/hive/clustersync.go

```diff
@@ -107,3 +137,31 @@ func (r *ReconcileHiveConfig) deployClusterSync(hLog log.FieldLogger, h resource
 	hLog.Info("all clustersync components successfully reconciled")
 	return nil
 }
+
+func hasStatefulSetSpecChanged(existingClusterSyncStatefulSet, newClusterSyncStatefulSet *appsv1.StatefulSet, hLog log.FieldLogger) bool {
+	// hash doesn't exist, assume the spec has changed.
+	if existingClusterSyncStatefulSet == nil {
+		hLog.Debug("existing clustersync statefulset is nil, so the spec didn't change.")
+		return false
+	}
+
+	if existingClusterSyncStatefulSet.Annotations == nil {
+		hLog.Debug("existing clustersync statefulset Annotations is nil, assuming spec changed.")
+		return true
+	}
+
+	existingClusterSyncStatefulSetSpecHash, found := existingClusterSyncStatefulSet.Annotations[hiveClusterSyncStatefulSetSpecHashAnnotation]
+	if !found {
+		hLog.Debug("existing clustersync statefulset Annotations doesn't contain the spec hash, assuming spec changed.")
+		return true
+	}
+
+	newClusterSyncStatefulSetSpecHash := newClusterSyncStatefulSet.Annotations[hiveClusterSyncStatefulSetSpecHashAnnotation]
+	if existingClusterSyncStatefulSetSpecHash != newClusterSyncStatefulSetSpecHash {
+		hLog.Debug("existing clustersync statefulset Spec changed")
+		return true
+	}
```

Comment:
what if we stored 2 hashed,
- hash for entire spec
- hash for mutable spec parts like replicas, template, updateStrategy

and we delete+create when hash(spec) changed and hash(mutable spec) remain unchanged?

this would reduce the times where we delete and recreate instead of rolling out the changes...

Reply:
That's an interesting idea.

Is it worth having to manually determine / maintain the list of mutable parts of the statefulset spec?

Since the deletes aren't ultimately destructive (we setup a new sts), and since the spec values don't change all that often, we're basically optimizing for not restarting the hive-clustersync pod as much (which we don't do that often now).

I'm down to do the work, not sure if it's worth it, though. Typically simpler logic is easier to maintain.

@dgoodwin What is your preference?